packages/nextra/src/compile.ts

```diff
@@ -52,45 +54,57 @@ export async function compileMdx(
     | 'defaultShowCopyCode'
     | 'readingTime'
     | 'latex'
+    | 'codeHighlight'
   > & {
     mdxOptions?: LoaderOptions['mdxOptions'] &
       Pick<ProcessorOptions, 'jsx' | 'outputFormat' | 'format'>
   } = {},
-  filePath = ''
+  filePath = '',
+  useCachedCompiler = false
```

Comment:
For dev and prod builds, when called from the loader all options here will not change. Hence we can just reuse the compiler.